app/controllers/checkout_controller.rb

```diff
@@ -58,6 +61,23 @@ def update
 
   private
 
+  def set_default_bill_address
+    if params[:order][:default_bill_address] == 'YES'
+      new_bill_address = @order.bill_address.clone
+      spree_current_user.set_bill_address(new_bill_address)
+      @order.customer.update_attribute(:bill_address, new_bill_address) unless @order.customer.bill_address
+    end
+
+  end
+
+  def set_default_ship_address
+    if params[:order][:default_ship_address] == 'YES'
+      new_ship_address = @order.ship_address.clone
+      spree_current_user.set_ship_address(new_ship_address)
+      @order.customer.update_attribute(:ship_address, new_ship_address) unless @order.customer.ship_address
```

Comment:
As we discussed before, if the customer was set an address from the admin interface, we won't change the address here. Because here is to set the default one, only when the customer doesn't have one.